From 500d1f0656267910ff142c19414fbb42f95005f1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 26 Nov 2020 08:37:24 +0200
Subject: [PATCH] [SW-22029] update status for sp sram init failure

Currently we perform initialization of scratchpad SRAM of 64KB as part
of init using memcopy block and we bailout if init is not completed
within max timeout defined. This patch add changes to update SW status
in case initialization of scratchpad SRAM is not completed within max
timeout.

Change-Id: I9220759d6736713e1d087e6a804042f6384c233f
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 33577e6..979f30f 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -96,8 +96,11 @@
 		 * seconds sleep), else bailout.
 		 */
 		retry--;
-		if (!retry)
+		if (!retry) {
+			arcboot_rom_status_update(
+					PROGRESS_INIT_SP_SRAM_NOT_COMPLETED);
 			break;
+		}
 
 		udelay(HL_INIT_SP_SRAM_UDELAY);
 	}
diff --git a/include/arcboot.h b/include/arcboot.h
index 80b60e6..faffd00 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -84,35 +84,36 @@
 
 enum arcboot_rom_progress {
 	PROGRESS_INIT,
+	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED,
 	PROGRESS_REVOKE_LOC,
 	PROGRESS_REVOKE_LOC_0,
 	PROGRESS_REVOKE_LOC_1,
-	PROGRESS_KEY_LOC,
-	PROGRESS_KEY_LOC_0,			/* 5 */
+	PROGRESS_KEY_LOC, 			/* 5 */
+	PROGRESS_KEY_LOC_0,
 	PROGRESS_KEY_LOC_1,
 	PROGRESS_VER_LOC,
 	PROGRESS_VER_LOC_0,
-	PROGRESS_VER_LOC_1,
-	PROGRESS_BRINGUP_PCI_PRI,		/* 10 */
+	PROGRESS_VER_LOC_1,			/* 10 */
+	PROGRESS_BRINGUP_PCI_PRI,
 	PROGRESS_BRINGUP_PCI_PRI_DMA,
 	PROGRESS_BRINGUP_PCI_PRI_SECURITY,
 	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG,
-	PROGRESS_BRINGUP_PCI_SEC,
-	PROGRESS_BRINGUP_PCI_SEC_DMA,		/* 15 */
+	PROGRESS_BRINGUP_PCI_SEC,		/* 15 */
+	PROGRESS_BRINGUP_PCI_SEC_DMA,
 	PROGRESS_BRINGUP_PCI_SEC_SECURITY,
 	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG,
 	PROGRESS_PCIE_IN_RECOVERY,
-	PROGRESS_PPBOOT_START,
-	PROGRESS_PPBOOT_FINISH,			/* 20 */
+	PROGRESS_PPBOOT_START,			/* 20 */
+	PROGRESS_PPBOOT_FINISH,
 	PROGRESS_BRINGUP_RAM_PRI,
 	PROGRESS_BRINGUP_RAM_PRI_DMA,
 	PROGRESS_BRINGUP_RAM_PRI_SECURITY,
-	PROGRESS_BRINGUP_RAM_SEC,
-	PROGRESS_BRINGUP_RAM_SEC_DMA,		/* 25 */
+	PROGRESS_BRINGUP_RAM_SEC,		/* 25 */
+	PROGRESS_BRINGUP_RAM_SEC_DMA,
 	PROGRESS_BRINGUP_RAM_SEC_SECURITY,
 	PROGRESS_ARCBOOT_RAM_START,
 	PROGRESS_ARCBOOT_RECOVERY,
-	PROGRESS_ARCBOOT_RECOVERY_DMA
+	PROGRESS_ARCBOOT_RECOVERY_DMA		/* 30 */
 };
 
 enum security_location {
